### PR TITLE
qlikview-client

### PR DESCRIPTION
**Pull request by andriy-bulynko**

### CODE REVIEW

**Review by bkuschel — APPROVED** [review]

File: manifests/base/resources/qlikview-client/transformers/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+- ../../../transformers
+transformers:
+- ../../../../../.operator/transformers
+- ../../../../../configuration/transformers
+#Disable all transformers except these
+patches:
+  - target:
+      labelSelector: >-
+        key notin (
+          ingressClass,
```
Comment (ffoysal) [inline manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: dont need ingressAuthUrl ??

**Comment by ashwathishiva** [comment]
Don't we also need manifests/base/resources/qlikview-client/patches/deployment.yaml?

File: manifests/base/resources/qlikview-client/secrets/selectivepatch.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  name: qlikview-client-component-secrets
+enabled: true
+patches:
+- target:
+    kind: Secret
+  patch: |-
+    - op: add
+      path: /stringData
+      value:
+        mongoDbUri: mongodb://$(PREFIX)-mongodb:27017/PDS
```
Comment (MoKassem) [inline manifests/base/resources/qlikview-client/secrets/selectivepatch.yaml]: `qlikview-client` doesn't have a mongo dependency. So we may need to rmv that plz.

File: manifests/base/resources/qlikview-client/transformers/kustomization.yaml (hunk 1 above)
Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: Not sure about ingress auth...
```
MacBook-Pro:qlikview-client dlx$ pwd
/tmp/.chartcache/qliksense/charts/qlikview-client
MacBook-Pro:qlikview-client dlx$ helm template .
---
# Source: qlikview-client/templates/service.yaml
apiVersion: v1
kind: Service
metadata:
  name: release-name-qlikview-client
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    release: release-name
    heritage: Tiller
  annotations:
    prometheus.io/scrape: "true"
    prometheus.io/port: "9180"
spec:
  type: ClusterIP
  ports:
    - port: 9003
      targetPort: 9003
      protocol: TCP
      name: qlikview-client
  selector:
    app: qlikview-client
    release: release-name

---
# Source: qlikview-client/templates/deployment.yaml
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: release-name-qlikview-client
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    release: release-name
    heritage: Tiller
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: qlikview-client
        release: release-name
    spec:
      containers:
        - name: qlikview-client
          image: qlik-docker-qsefe.bintray.io/qlikview-client:12.50.2055
          imagePullPolicy:
          ports:
            - containerPort: 9003
          livenessProbe:
            httpGet:
              path: /health
              port: 9003
      imagePullSecrets:
        - name: artifactory-docker-secret

  selector:
    matchLabels:
      app: qlikview-client
      release: release-name

---
# Source: qlikview-client/templates/ingress.yaml
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: release-name-qlikview-client
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    release: release-name
    heritage: Tiller
  annotations:
    nginx.ingress.kubernetes.io/rewrite-target: "/opendoc.htm"
    kubernetes.io/ingress.class: nginx
spec:
  rules:
    - http:
        paths:
          - path: /qv/opendoc.htm
            backend:
              serviceName: release-name-qlikview-client
              servicePort: 9003
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: release-name-qlikview-client-ajax
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    release: release-name
    heritage: Tiller
  annotations:
    kubernetes.io/ingress.class: nginx
spec:
  rules:
    - http:
        paths:
          - path: /QvAjaxZfc
            backend:
              serviceName: release-name-qlikview-client
              servicePort: 9003

MacBook-Pro:qlikview-client dlx$
```

File: manifests/base/resources/qlikview-client/secrets/selectivepatch.yaml (hunk 2 above)
Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/secrets/selectivepatch.yaml]: @MoKassem  On it and done.

File: manifests/base/resources/qlikview-client/transformers/kustomization.yaml (hunk 1 above)
Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: @ffoysal Are these 2 ingresses unauthenticated?

**Comment by andriy-bulynko** [comment]
> Don't we also need manifests/base/resources/qlikview-client/patches/deployment.yaml?

@ashwathishiva I'm not sure... There is nothing in the deployment to patch... There are no ENV variables.. Or am I missing something?

File: manifests/base/resources/qlikview-client/transformers/kustomization.yaml (hunk 1 above)
Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: @ffoysal Actually, this is what the 2 ingresses look like after our "kustomizations":
```
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: qlik-nginx
    nginx.ingress.kubernetes.io/auth-url: http://qliksense-edge-auth.default.svc.cluster.local:8080/v1/auth
    nginx.ingress.kubernetes.io/rewrite-target: /opendoc.htm
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    heritage: Tiller
    release: qliksense
  name: qliksense-qlikview-client
  namespace: default
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: qliksense-qlikview-client
          servicePort: 9003
        path: /qv/opendoc.htm
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: qlik-nginx
    nginx.ingress.kubernetes.io/auth-url: http://qliksense-edge-auth.default.svc.cluster.local:8080/v1/auth
  labels:
    app: qlikview-client
    chart: qlikview-client-1.0.12
    heritage: Tiller
    release: qliksense
  name: qliksense-qlikview-client-ajax
  namespace: default
spec:
  rules:
  - http:
      paths:
      - backend:
          serviceName: qliksense-qlikview-client
          servicePort: 9003
        path: /QvAjaxZfc
---
```

Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: @ffoysal What does the `ingressAuthUrl` transformer do?

Comment (andriy-bulynko) [reply manifests/base/resources/qlikview-client/transformers/kustomization.yaml:15]: @ffoysal I just tried inserting the `ingressAuthUrl` transformer and re-running `kuz` and it gave me the same output as without it...

**Pull request merged**